### PR TITLE
test: fix Cypress test flakes

### LINKED ISSUE #4739: Website tests flake: ESOCKETTIMEDOUT
## Overview

Cypress flake in CircleCI! Hooray!

```plaintext
  0 passing (31s)
  1 failing

  1) Index
       has no accessibility issues detected by aXe:
     CypressError: `cy.visit()` failed trying to load:

http://localhost:3000/

We attempted to make an http request to this URL but the request failed without a response.

We received this error at the network level:

  > Error: ESOCKETTIMEDOUT

```

Example: https://github.com/typescript-eslint/typescript-eslint/runs/5648789441?check_suite_focus=true

It started happening after https://github.com/typescript-eslint/typescript-eslint/pull/4710. I'd specifically kept us at a lower cypress version in https://github.com/typescript-eslint/typescript-eslint/pull/4362 because of https://github.com/cypress-io/cypress/issues/19299, a different piece of flake. Lovely.

I'll revert #4710 and pin us to the old version of Cypress for now.

### PR DESCRIPTION
**Pull request by Josh-Cena**
## PR Checklist

-   [x] Addresses an existing open issue: fix #4739
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

### CODE REVIEW

File: .github/actions/prepare-install/action.yml [hunk 1]
```diff
@@ -31,7 +31,9 @@ runs:
     - uses: actions/cache@v3
       id: yarn-cache
       with:
-        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        path: |
+          ${{ steps.yarn-cache-dir-path.outputs.dir }}
```
Comment (Josh-Cena) [inline .github/actions/prepare-install/action.yml:35]: This is essentially a subpath of `~/.cache`. I don't know if there would be any adverse effects if we simply cache the entire `~/.cache`, or if I should constrain it to certain subpaths.

**Comment by Josh-Cena** [comment]
I have to `yarn cypress install`, although I did persist the cache. I don't know if that's fully necessary, but it at least works this way, and doesn't take much time anyway.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -193,11 +193,15 @@ jobs:
...
 
+      - name: Install Cypress
+        run: yarn cypress install
+
       - name: Cypress run
         uses: cypress-io/github-action@v2
         with:
           project: ./packages/website
           start: yarn start
+          wait-on: http://localhost:3000
```
Comment (Josh-Cena) [inline .github/workflows/ci.yml:204]: Side fix: this should be able to fix the ESOCKETTIMEOUT error.

**Comment by Josh-Cena** [comment]
@JoshuaKGoldberg Do you want to try bumping the Cypress version in this PR? Doesn't seem that flaky now.

**Comment by Josh-Cena** [comment]
> `Error: connect ECONNREFUSED 127.0.0.1:3000`

😨

**Comment by bradzacher** [comment]
maybe we don't need the `wait-on`? The action passed without it!

**Comment by Josh-Cena** [comment]
Right... I'm still worried that ESOCKETTIMEDOUT will come back, particularly if we stop pinning our Cypress version.

**Review by bradzacher — APPROVED** [review]
image

image

**Comment by JoshuaKGoldberg** [comment]
yolo

**Pull request merged**